plugins/deisa/deisa.cxx

```diff
@@ -0,0 +1,205 @@
+				each(value, [&](PC_tree_t key_map, PC_tree_t value_map) { m_deisa_arrays.emplace(to_string(key_map), ctx.datatype(value_map)); });
+			} else if (key == "time_step") {
+				m_time_step = to_string(value);
+			} else if (key == "map_in") {
+				//
+			} else if (key == "logging" || key == "init_on") {
+				//
+			} else {
+				throw Config_error{key_tree, "Unknown key in Deisa file configuration: '{}'", key};
+			}
+		});
+
+		// plugin init
+		PC_tree_t init_tree = PC_get(conf, ".init_on");
+		if (!PC_status(init_tree)) {
+			ctx.callbacks().add_event_callback([&](const std::string&) { init_deisa(ctx); }, to_string(init_tree));
+		} else {
+			throw Config_error{conf, "Deisa plugin requires init_on key "};
+		} // TODO: replace with try/catch when #480 is fixed.
+
+		// map_in
+		PC_tree_t map_in = PC_get(conf, ".map_in");
+		if (!PC_status(map_in)) {
+			each(map_in, [&](PC_tree_t key_map, PC_tree_t value_map) { // TODO: when #481 is fixed, use `or_none` variant.
+				ctx.callbacks().add_data_callback(
+					[&, deisa_array_name = to_string(value_map)](const std::string&, const Ref& data_ref) {
+						try {
+							assert(m_bridge != py::none());
+							assert(hasattr(m_bridge, "publish_data"));
+							py::object publish_data = m_bridge.attr("publish_data");
+#ifdef NDEBUG
+							publish_data(to_python(data_ref), deisa_array_name.c_str(), m_time_step.to_long(ctx), "debug"_a = false);
+#else
+							publish_data(to_python(data_ref), deisa_array_name.c_str(), m_time_step.to_long(ctx), "debug"_a=true);
+#endif
+						} catch (const std::exception& e) {
+							throw Plugin_error("While publishing data. Caught exception: {}", std::string(e.what()));
+						} catch (...) {
+							throw Plugin_error("While publishing data.");
+						}
+					},
+					to_string(key_map)
+				);
+			});
+		}
+	}
+
+	~deisa_plugin() noexcept override
+	{
+		try {
+			// call bridge release() so that we can clear things before destructor is called (if it is ever called !).
+			assert(hasattr(m_bridge, "release"));
+			m_bridge.release();
+			m_bridge = py::none();
+			py::finalize_interpreter();
+		} catch (const std::exception& e) {
+			context().logger().error("Exception in destructor, caught exception {}", e.what());
+		} catch (...) {
+			context().logger().error("Exception in destructor");
+		}
+	}
+
+private:
+	/**
+	 * Check that the PDI plugin is compatible with Deisa's python Bridge (i.e, that the python API is what it should be).
+	 * The plugin is compatible if DEISA_COMPATIBLE_VERSION == deisa.__version__.__version__ (python)
+	 */
+	void check_compatibility()
+	{
+		py::module deisa = py::module::import("deisa.__version__");
+		const auto python_library_version = py::str(deisa.attr("__version__")).cast<std::string>();
+		if (python_library_version != DEISA_COMPATIBLE_VERSION) {
+			throw Plugin_error(
+				"Deisa PDI plugin is expecting Deisa python version " + std::string(DEISA_COMPATIBLE_VERSION) + " but found version "
+				+ python_library_version
```

Comment:
It would be better to use the fmt support here:
```suggestion
				"Deisa PDI plugin is expecting Deisa python version {} but found version {}", DEISA_COMPATIBLE_VERSION, python_library_version
```